### PR TITLE
Improve index page of Jekyll documentation 

### PR DESCRIPTION
**Pull request by AninditaBasu**
This is a 🔦 documentation change.

- [x] I read the contributing document at https://jekyllrb.com/docs/contributing/

  - [ ] I added tests (if it's a bug, feature or enhancement)
  - [x] I've adjusted the documentation (if it's a feature or enhancement)
  - [ ] The test suite passes locally (run `script/cibuild` to verify this)

## Summary

1. Moved the prerequisites link to the top of the page (instead of the bottom) so it's available right away. 
2. Removed ableist language ('simply') and reworded the introductory paragraph to use less jargon.

## Context

I came to the doc site to see how to use Jekyll, but couldn't find the prerequisites easily, either on the Quickstart page or the Installation page.

### CODE REVIEW

**Comment by ashmaroli** [comment]
@AninditaBasu Thank you for the contribution. But I would like to shed some light on your changes:
- `Jekyll is a simple, extendable, static site generator.` *simple* here is an adjective and not a linguistic sugar.
- *layout templates* are not a requirement per se. Even if there are no layout templates, a static site will still be generated.
- `/docs/installation/` is not equivalent to `/docs/installation`. The former would infer `/docs/installation/index.html` while the latter can mean `/docs/installation.html` based on server settings.

**Comment by AninditaBasu** [comment]
- um, `simple` is subjective (while `extendable` isn't) so...
I'll delete it if you still don't agree.
- I changed it to 'it uses layouts' (is 'it churns through layouts' on the currently live doc site)
- sorry, this was a mistake on my part. restored.

**Comment by ashmaroli** [comment]
I don't see a need to enforce *simple*.. so your call :man_shrugging: 

However, I would like the line-lengths to follow existing (or surrounding) styles to improve readability of the source document **in a local text-editor**. Its *looks weird* that some lines are capped at around 80 chars while some are at around 70 while some are so long that it folds over the next line.

I understand that it is hard to follow this when editing via the GitHub interface, but would prefer it be maintained as possible.

**Comment by AninditaBasu** [comment]
:) I can live without _simple_

Am sorry about the line length issue. Was editing via the GitHub interface, so didn't realise the wrapping would look weird on a local editor. Have tried to fix it. Does it look okay now? 

Thank you for your patience.

**Comment by ashmaroli** [comment]
> Am sorry about the line length issue. Was editing via the GitHub interface

Don't worry about it. I've pushed a commit to align the sentences properly.

**Comment by ashmaroli** [comment]
Thank you @AninditaBasu for improving the docs.

@jekyllbot: merge +doc

**Comment by mattr-** [comment]
Thank you for removing our ableist language and improving our documentation. ❤
